### PR TITLE
Issue #696: Fix rake rollbar:test and update for rubocop and rspec

### PR DESCRIPTION
**Pull request by waltjones**
https://github.com/rollbar/rollbar-gem/issues/696

Refactor rake rollbar:test:
- [x] Refactors rake_tasks.rb to use a module callable from Rspec
- [x] Refactors rake_tasks.rb to pass rubocop
- [x] Adds RSpec examples for primary use cases

Fix https://github.com/rollbar/rollbar-gem/issues/696:
- [x] Run the verification test using a clone of the default Rails app.
- [x] Resolve RSpec issues with Rails 4.x

The root cause of the issue, "ActionController::RoutingError (No route matches [GET] "/verify")" when running `rake rollbar:test:`, is that when Spring is running in the background and `config.cache_classes == false` in the current Rails environment, routes are reset on each request and the new test route can't be found.

The issue was initially difficult to reproduce because, even if Spring is already running, `bin/rake` will use Spring and reproduce the error. However, `bundle exec rake` does not use Spring and therefore does not reproduce the error.

In short, to see this error you must have Spring running, have `config.cache_classes == false`, and use `bin/rake` (or at least a way of invoking rake that does use Spring.)

Several techniques were considered to work around this issue. Neither the middleware chain (`ActionDispatch::Reloader` does the actual work resetting the routes) nor the `cache_classes` flag can be changed after Rails is loaded.

Cloning the app allows the config to be updated without modifying the default app. While changes to the default app config could be safely reverted by using `ensure` sections, the code is shorter and cleaner when using the cloned app.

### CODE REVIEW

**Comment by rokob** [comment]
I also looked at this earlier but apparently I didn't leave a comment. But yeah this looks great.

**Comment by waltjones** [comment]
@ArturMoczulski , @rokob , thank you!
